python/lsst/sims/maf/batches/timeBatch.py

```diff
@@ -100,21 +104,102 @@ def intraNight(colmap=None, runName='opsim', nside=64, sqlConstraint=None):
 
 
-def interNight(colmap=None, runName='opsim', nside=64):
-    """Generate a set of statistics about the gaps between nights of observations.
+def interNight(colmap=None, runName='opsim', nside=64, extraSql=None, extraMetadata=None):
+    """Generate a set of statistics about the spacing between nights with observations.
+
+    Parameters
+    ----------
+    colmap : dict or None, opt
+        A dictionary with a mapping of column names. Default will use OpsimV4 column names.
+    runName : str, opt
+        The name of the simulated survey. Default is "opsim".
+    nside : int, opt
+        Nside for the healpix slicer. Default 64.
+    extraSql : str or None, opt
+        Additional sql constraint to apply to all metrics.
+    extraMetadata : str or None, opt
+        Additional metadata to use for all outputs.
+
+    Returns
+    -------
+    metricBundleDict
+    """
 
-     Parameters
-     ----------
-     colmap : dict or None opt
-         A dictionary with a mapping of column names. Default will use OpsimV4 column names.
-     runName : str, opt
-         The name of the simulated survey. Default is "opsim".
-     nside : int, opt
-         Nside for the healpix slicer. Default 64.
+    if colmap is None:
+        colmap = ColMapDict('opsimV4')
 
-     Returns
-     -------
-     metricBundleDict
-     """
-    pass
+    bundleList = []
 
+    metadata = extraMetadata
+    if extraSql is None:
+        extraSql = ''
+    if extraSql is not None and len(extraSql) > 0:
+        if metadata is None:
+            metadata = extraSql
+
+    filterlist = ('u', 'g', 'r', 'i', 'z', 'y', 'all')
+    colors = {'u': 'cyan', 'g': 'g', 'r': 'y', 'i': 'r', 'z': 'm', 'y': 'b', 'all': 'k'}
+    filterorder = {'u': 1, 'g': 2, 'r': 3, 'i': 4, 'z': 5, 'y': 6, 'all': 0}
+
+    displayDict = {'group': 'InterNight', 'subgroup': 'Night gaps', 'caption': None, 'order': 0}
+    bins = np.arange(1, 20.5, 1)
+    metric = metrics.NightgapsMetric(bins=bins, nightCol=colmap['night'], metricName='DeltaNight Histogram')
+    slicer = slicers.HealpixSlicer(nside=nside, latCol=colmap['dec'], lonCol=colmap['ra'],
+                                   latLonDeg=colmap['raDecDeg'])
+    plotDict = {'bins': bins, 'xlabel': 'dT (nights)'}
+    displayDict['caption'] = 'Histogram of the number of nights between consecutive visits to a ' \
+                             'given point on the sky, considering separations between %d and %d.' \
+                             % (bins.min(), bins.max())
+    plotFunc = plots.SummaryHistogram()
+    bundle = mb.MetricBundle(metric, slicer, extraSql, plotDict=plotDict,
+                             displayDict=displayDict, metadata=metadata, plotFuncs=[plotFunc])
+    bundleList.append(bundle)
+
+    standardStats = standardSummary()
+    subsetPlots = [plots.HealpixSkyMap(), plots.HealpixHistogram()]
+
+    # Median inter-night gap (each and all filters)
+    metric = metrics.InterNightGapsMetric(metricName='Median Inter-Night Gap')
```

Comment:
Although `np.median` is the default `reducFunc` for this metric should we still set it in this call for readability? I'm fine either way, but I had to go make sure it was the default in the metric.

Reply:
Good idea.